### PR TITLE
[BTS-275] Applying theme props to Checkbox

### PR DESCRIPTION
**Pull request by ddsilva**

### CODE REVIEW

File: components/Checkbox/Checkbox.jsx [hunk 1]
```diff
@@ -38,69 +46,110 @@ const HiddenCheckbox = styled(HiddenInput).attrs({
...
+      }
+
+      :checked + ${CheckIcon} {
+        background-color: ${neutral[500]};
+      }
+    `}
 
     :hover + ${CheckIcon} {
-      box-shadow: none;
+      ${shadow(0)}
```
Comment (ggdaltoso) [inline components/Checkbox/Checkbox.jsx]: Isn't 0 the default elevation value? If yes, is it necessary here?

File: components/Checkbox/Checkbox.jsx [hunk 2]
```diff
@@ -18,9 +18,17 @@ const CheckboxWrapper = styled.div`
...
+    },
+  }) => neutral[700]};
   font-size: 16px;
-  margin: 0 0 0 10px;
+  margin: ${({
+    theme: {
+      spacing: { xsmall },
+    },
+  }) => `0 0 0 ${xsmall}px`};
 `;
```
Comment (allyssonsantos) [inline components/Checkbox/Checkbox.jsx]: What do you think to put the color and margin callbacks together?

```suggestion
const CheckboxLabel = styled(Label)`
  ${({ theme: { colors: { neutral }, spacing: { xsmall } } }) => `
    color: ${neutral[700]};
    margin: 0 0 0 ${xsmall}px;
  `}
  font-size: 16px;
`;
```

File: components/Checkbox/Checkbox.jsx [hunk 3]
```diff
@@ -38,69 +46,110 @@ const HiddenCheckbox = styled(HiddenInput).attrs({
...
     justify-content: center;
-    margin-top: 2px;
+    margin-top: ${({
+      theme: {
+        spacing: { xxxsmall },
+      },
+    }) => `${xxxsmall}px`};
     transition: all 0.2s ease-in-out;
     width: ${CHECKBOX_SIZE};
+
```
Comment (allyssonsantos) [inline components/Checkbox/Checkbox.jsx]: And here too.

File: components/Checkbox/Checkbox.jsx [hunk 4]
```diff
@@ -38,69 +46,110 @@ const HiddenCheckbox = styled(HiddenInput).attrs({
 
   + ${CheckIcon} {
     align-items: center;
-    background-color: ${Colors.WHITE};
     border-radius: 2px;
-    border: 2px solid ${Colors.BLACK['400']};
     box-sizing: border-box;
     color: transparent;
     display: flex;
     font-size: 16px;
```
Comment (alan-oliv) [inline components/Checkbox/Checkbox.jsx]: There's a reason not to use baseFontSize from theme here?

File: components/Checkbox/Checkbox.jsx (hunk 2 above)
Comment (alan-oliv) [reply components/Checkbox/Checkbox.jsx]: And baseFontSize too (if there's not a reason)

**Pull request merged**